notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java

```diff
@@ -275,6 +275,7 @@ public Page updateNote(Page note, PageUpdateType type, Identity userIdentity) th
       }
     }
     Page updatedPage = updateNote(note, type);
+    Utils.broadcast(listenerService, "note.updated", userIdentity.getUserId(), updatedPage);
```

Comment:
Shouldn't we trigger/centralize this event in method `public Page updateNote(Page note, PageUpdateType type)`, WDYT ?

Reply:
I used the updatePage function that is called in the rest layer in all update actions while the above method is used only in the service layer.
And the important reason is that i need the `userIdentity` which is not available in this method.

Reply:
Yep, this is not the good pattern to use for event triggering .... When the update id made by a batch update (auto import when server starts up by example), the listener should be triggered as well if if no user had auctioned the update.

Reply:
Ok but without the userId the listener is useless
https://github.com/Meeds-io/notes/pull/843/files#diff-3198ed87ff299273c36b60c054d7cfdef2b2eada4a3f90bd54d51fe498237cb7R61-R62

Reply:
Is it really?

```java
    Page note = event.getData();
    String username = event.getSource();
    long creatorId = getPosterId(username);
    long audienceId = getStreamOwnerId(note.getWikiOwner(), username);

    Set<TagName> tagNames = tagService.detectTagNames(note.getContent());
    tagService.saveTags(new TagObject(NOTES_METADATA_OBJECT_TYPE, note.getId(), note.getParentPageId()),
                        tagNames,
                        audienceId,
                        creatorId);
    indexingService.reindex(WikiPageIndexingServiceConnector.TYPE, note.getId());
```

The listener should:
* update Tags
* reindex the page

**independently** whether the action is made **by a user** or **by a batch process**, those updates has to be made.

Shouldn't we delete the `indexingService.reindex` since the note page is automatically reindexed when a metadata item is modified (triggered using listener `org.exoplatform.wiki.service.listener.MetadataItemModified`) ?

Reply:
I didn't understand,
```
 tagService.saveTags(new TagObject(NOTES_METADATA_OBJECT_TYPE, note.getId(), note.getParentPageId()),
                        tagNames,
                        audienceId,
                        creatorId
```); 
In the createTags isn't the userIdentityId mandatory ? 
how i can get it without passing it as parameter in the event?

Reply:
you can use the last updater or the creator, it's not really important information that will be used after saving the associated Tags to a note, the most important information is the `audienceId`

Reply:
> There is no creator or updater property in the note page and the `page.getAuthor()` is always null because there is no author property in `pageEntity`

I let you find a solution, for my part the flow and layers are important to preserve coherence and information consistency.